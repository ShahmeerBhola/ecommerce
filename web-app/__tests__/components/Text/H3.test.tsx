import React from 'react';
import renderer from 'react-test-renderer';
import H3 from '../../../components/Text/H3';

describe('H3 component', () => {
  test('it renders properly with default color', () => {
    const tree = renderer.create(<H3>Header</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default color', () => {
    const tree = renderer.create(<H3 color="red-200">Header</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with className prop', () => {
    const tree = renderer.create(<H3 className="my-custom-class">Header</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with uppercase prop = false', () => {
    const tree = renderer.create(<H3 uppercase={false}>Header</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with uppercase prop = true', () => {
    const tree = renderer.create(<H3 uppercase>Header</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
