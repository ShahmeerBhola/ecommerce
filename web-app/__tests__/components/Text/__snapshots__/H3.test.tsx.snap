// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 component it renders properly with className prop 1`] = `
<h3
  className="font-primary font-bold text-xl my-custom-class uppercase text-gray-100"
>
  Header
</h3>
`;

exports[`H3 component it renders properly with default color 1`] = `
<h3
  className="font-primary font-bold text-xl uppercase text-gray-100"
>
  Header
</h3>
`;

exports[`H3 component it renders properly with non-default color 1`] = `
<h3
  className="font-primary font-bold text-xl uppercase text-red-200"
>
  Header
</h3>
`;

exports[`H3 component it renders properly with uppercase prop = false 1`] = `
<h3
  className="font-primary font-bold text-xl text-gray-100"
>
  Header
</h3>
`;

exports[`H3 component it renders properly with uppercase prop = true 1`] = `
<h3
  className="font-primary font-bold text-xl uppercase text-gray-100"
>
  Header
</h3>
`;
